expression/chunk_executor.go

```diff
@@ -53,6 +53,27 @@ func HasGetSetVarFunc(expr Expression) bool {
 	return false
 }
 
+// HasAssignSetVarFunc checks whether an expression contains SetVar function and assign a value
+func HasAssignSetVarFunc(expr Expression) bool {
```

Comment:
This check looks pretty ad-hoc, I prefer to push no condition down to Projection's child completely if `p.Exprs` contains `SetVar` or `GetVar`, because there seems like many hidden corner cases which would return wrong results. @zz-jason How do you think?

Reply:
If p.Exprs contains setvar and getvar and does not push down all the conditions, then many normal pushdowns will be ignored, which can have a large impact on performance in many cases. I think this kind of check just a subset of reject all pushdowns, which helps narrow down the problem without expanding the problem. @eurekaka  @zz-jason How do you think?